testnet/puppet-chain/src/main.rs

```diff
@@ -92,38 +94,76 @@ async fn main() -> http_types::Result<()> {
 
     // Start a loop in a separate thread, generating new blocks
     // on a given frequence (coming from config).
+    let boot_height = num_blocks;
+    let block_height_reader = Arc::new(Mutex::new(num_blocks));
+    let block_height_writer = block_height_reader.clone();
     let conf = config.clone();
     thread::spawn(move || {
-        let miner_address = conf.neon.miner_address.clone();
+        let miner_address = conf.network.miner_address.clone();
 
         loop {
-            println!("Generating block {}", num_blocks);
+            let delay = {
+                let mut block_height = block_height_writer.lock().unwrap();
+                let block_time = conf.get_block_time_at_height(*block_height - num_blocks);
+                println!(
+                    "Generating block {} (block_time: {}ms)",
+                    block_height, block_time
+                );
+                *block_height += 1;
+                block_time
+            };
             async_std::task::block_on(async {
                 generate_blocks(1, miner_address.clone(), &conf).await;
             });
-            thread::sleep(block_time);
-            num_blocks += 1;
+
+            thread::sleep(Duration::from_millis(delay));
         }
     });
 
     // Open up a TCP connection and create a URL.
-    let bind_addr = config.neon.rpc_bind.clone();
+    let bind_addr = config.network.rpc_bind.clone();
     let listener = TcpListener::bind(bind_addr).await?;
     let addr = format!("http://{}", listener.local_addr()?);
     println!("Listening on {}", addr);
 
     // For each incoming TCP connection, spawn a task and call `accept`.
     let mut incoming = listener.incoming();
+    let mut buffered_requests = VecDeque::new();
     while let Some(stream) = incoming.next().await {
+        let block_height = block_height_reader.lock().unwrap();
+        let effective_block_height = *block_height - boot_height;
+        let should_ignore_txs = config.should_ignore_transactions(effective_block_height);
+
         let stream = stream?;
         let addr = addr.clone();
-        let config = config.clone();
 
-        task::spawn(async move {
-            if let Err(err) = accept(addr, stream, &config).await {
-                eprintln!("{}", err);
+        if should_ignore_txs {
+            // Returns ok
+            println!("will buffer request from {}", stream.peer_addr()?);
+            async_h1::accept(&addr, stream.clone(), |_| async {
+                Ok(Response::new(StatusCode::Ok))
+            })
+            .await?;
+            // Enqueue request
+            buffered_requests.push_back((addr, stream));
```

Comment:
This approach feels fragile to me.  It looks like you're accepting all inbound connections in the `should_ignore_txs` branch, and then just...idling them?  What happens if the client times out?  What happens if you have enough inbound connections during the block that you run out of file descriptors?  If the point of doing this is just to buffer transactions, but not to delay other RPCs to bitcoind (like `getnetworkinfo` or `getchaininfo`), then why not just accept and buffer up the transactions, and then when the next block gets generated, send them to the bitcoind node then?  Then, downstream clients won't be kept waiting.  You could even immediately reply to downstream clients with the txid, even though you've buffered up the transactions.

Reply:
No, we're not idling them, we are blindly returning some `200s` (see line `144`).